Python/initconfig.c

```diff
@@ -1778,12 +1800,15 @@ config_init_tracemalloc(PyConfig *config)
 static PyStatus
 config_init_int_max_str_digits(PyConfig *config)
 {
-    int maxdigits;
-    int valid = 0;
+    int32_t maxdigits;
+    bool valid = 0;
+
+    /* default to unconfigured, _PyLong_InitTypes() does the rest */
+    config->int_max_str_digits = -1;
```

Comment:
The PyConfig API gives the highest priority to PyConfig: the caller should be able to override the default value. Example:

```
PyConfig config;
PyConfig_InitPythonConfig(&config);
config.int_max_str_digits = 1234;  // Python must use this value
```

The default is already initialized in _PyConfig_InitCompatConfig(), I don't see the need to reset the value to its default here. Wouild you mind to add a test? In Programs/_testembed.c: set a value in test_init_from_config() and then update Lib/test/test_embed.py.